### DISCUSSION TITLE
SwiftUI views inside package

### DISCUSSION THREAD

**Discussion by ariskox** [skiptools/skip#179] [Q&A] [answered]
Hello,

I have created a multi-module app and  I'm trying to add a SwiftUI view inside a module. The code is:

`
import SwiftUI

public struct ProfileView: View {
    public init() {}

    public var body: some View {
        VStack {
            Image(systemName: "profile")
                .resizable()
                .frame(width: 100, height: 100)
                .clipShape(Circle())
                .padding()
            Text("This is a test")
                .font(Font.headline)
                .padding()

        }
    }
}
`
When compiling this, I get the following warning inside the file:

> Skip is unable to match this API call to determine whether it results in a View. Consider adding additional type information

But the program fails to compile the android version with the following issues:

![Screenshot 2024-07-26 at 5 27 48 PM](https://github.com/user-attachments/assets/5d637b01-5759-4584-be18-7bf4f9fa1d19)

The same code compiles successfully if I move it to the 'main module'.

How can we fix this?

Thanks

**Comment by marcprux** [comment, marked as answer]
I'm guessing that the module's target doesn't have SkipUI as a dependency, which is where the "Unresolved reference 'ui." error is coming from. Can you try editing your `Package.swift` and adding `SkipUI` as a dependency of your module?

If that doesn't help, can you post the contents of your `Package.swift` file?

**Reply by ariskox** [reply]
Thank you very much. I added SkipUI as a dependency and the issue is resolved!

Now, how about the following error?

`
Argument type mismatch: actual type is 'skip.ui.Shape', but 'skip.ui.ShapeStyle' was expected.
`

this is caused by adding the following to an image
'
.overlay(Circle().stroke(Color.white, lineWidth: 4))
'

**Reply by marcprux** [reply]
Can you try making the `overlay` call take an explicit closure? E.g.:

```
.overlay {
    Circle().stroke(Color.white, lineWidth: 4)
}
```

**Reply by ariskox** [reply]
It works correctly. Thank you
